### PR TITLE
Add UrlLib response buffer unit test

### PR DESCRIPTION
**Pull request by docEdub**
Needs https://github.com/BabylonJS/UrlLib/pull/13 to pass.

### CODE REVIEW

File: Tests/UnitTests/Scripts/tests.js [hunk 1]
```diff
@@ -177,6 +180,13 @@ describe("XMLHTTPRequest", function () {
             expect(xhr).to.have.property('responseText', 'var symlink_target_js = true;');
         });
     }
+
+    it("should load URL as array buffer", async function () {
+        const xhr = await createRequest("GET", "app:///Scripts/symlink_target.js", undefined, "arraybuffer");
+        var expected = Uint8Array.from("var symlink_target_js = true;".split(""), x => x.charCodeAt(0));
```
Comment (bghgary) [inline Tests/UnitTests/Scripts/tests.js:186]: What's the `split` for?

**Review by bghgary — APPROVED** [review]

File: Tests/UnitTests/Scripts/tests.js (hunk 1 above)
Comment (docEdub) [reply Tests/UnitTests/Scripts/tests.js:186]: Splits the string into individual characters fed as `x` into `x.charCodeAt(0)`. `Uint8Array.from("...")` without the character code conversion doesn't match.

Comment (bghgary) [reply Tests/UnitTests/Scripts/tests.js:186]: I'm not sure this is supported in Chakra, but if so, you can use https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder/encode.

Comment (docEdub) [reply Tests/UnitTests/Scripts/tests.js:186]: That gives `ReferenceError: TextEncoder is not defined` on Chakra and V8. I haven't tried it on JSC.

Comment (bghgary) [reply Tests/UnitTests/Scripts/tests.js:186]: Ah, yes, it does appear to be a DOM feature, so not part of JavaScript. Never mind.

Comment (bghgary) [reply Tests/UnitTests/Scripts/tests.js:186]: I guess another option is to add another file that is actually binary with a few bytes that you can just hard code here.

**Comment by bghgary** [comment]
Build is failing for Android JSC :-/

> 1 XMLHTTPRequest
> 10-04 19:10:29.717  3194  3212 I JsRuntimeHost:        should load URL as array buffer TypedArray.from requires its this argument subclass a TypedArray constructor   from@[native code]
> 10-04 19:10:29.717  3194  3212 I JsRuntimeHost:   app:///Scripts/tests.js:186:39

**Comment by docEdub** [comment]
> Build is failing for Android JSC :-/
> 
> > 1 XMLHTTPRequest
> > 10-04 19:10:29.717  3194  3212 I JsRuntimeHost:        should load URL as array buffer TypedArray.from requires its this argument subclass a TypedArray constructor   from@[native code]
> > 10-04 19:10:29.717  3194  3212 I JsRuntimeHost:   app:///Scripts/tests.js:186:39

Ya, I'm looking at it now. Works on Android API 33 but not the CI's API 27.

**Comment by bghgary** [comment]
That's weird. I don't see how API level has to do with the JS.

**Comment by docEdub** [comment]
> That's weird. I don't see how Android API level has to do with the JS.

Works on my local machine with API 27, too, so I'm not sure what's going on now.

> Are you testing with V8 or JSC?

I don't know. Whatever the default is.

**Comment by bghgary** [comment]
Default is V8 for Android. If you don't change to JSC, it probably won't repro.

**Comment by bghgary** [comment]
Put this in the `gradle.properties` file:
> jsEngine=JavaScriptCore

**Pull request merged**